src/Installers/Windows/WindowsHostingBundle/SharedFramework.wxs

```diff
@@ -40,7 +40,7 @@
                         Name="$(var.SharedFxRedistInstallerx64)"
                         Compressed="yes"
                         Vital="yes"
-                        InstallCondition="(NativeMachine=&quot;$(var.NativeMachine_x64)&quot;) AND (NOT OPT_NO_SHAREDFX OR OPT_NO_SHAREDFX=&quot;0&quot;)"
+                        InstallCondition="VersionNT64 AND NOT (NativeMachine=&quot;$(var.NativeMachine_arm64)&quot;) AND (NOT OPT_NO_SHAREDFX OR OPT_NO_SHAREDFX=&quot;0&quot;)"
```

Comment:
I still think this condition should be `VersionNT64 OR NativeMachine="34404"`. Any non-arm64 result would allow it to install and probably a similar OR condition for the arm64 install condition.

All `VersionNT64` does is tell you that you are on a 64-bit OS, if it's not arm64 it should be amd64, but you could also be on an IA64 platform in which case the x64 installer would not be valid.

Reply:
I guess I'm trying to keep this change to be the smallest possible, prior to the arm64 change this logic was simply
`VersionNT64 AND (NOT OPT_NO_SHAREDFX OR OPT_NO_SHAREDFX=&quot;0&quot;)` here in the release/6.0 source https://github.com/dotnet/aspnetcore/blob/release/6.0/src/Installers/Windows/WindowsHostingBundle/SharedFramework.wxs#L24

Or did something change in 7.0 where the 6.0 logic here is insufficient (i.e. was IA64 not an issue in 6.0?)

Reply:
IA64 is not something we support, it's just that on any unsupported OS, native machine image will always be set to unknown. I'm just paranoid we run into a case where we get a result that's neither unknown nor arm64, but not amd64 and we end up installing the x64 bits. There probably isn't a case like that right now, but the x64 logic now states any machine that's 64-bit and not arm64.

Reply:
Are you comfortable if I make the additional cleanup tweak in 8.0 and keeping the smaller change for 7.0 if IA64 isn't supported?

Reply:
I guess what I'm saying is any issues with the current situation would also show up for 6.0 on x64 too right?